src/andromede/main/main.py

```diff
@@ -0,0 +1,172 @@
+            yaml_lib = parse_yaml_library(file)
+            if yaml_lib.id in yaml_libraries:
+                raise ValueError(f"the identifier: {yaml_lib.id} is defined twice")
+            yaml_libraries[yaml_lib.id] = yaml_lib
+
+    todo = list(yaml_libraries.values())
+    did = list()
+    import_stack = []
+    output_lib = Library(port_types={}, models={})
+
+    while todo:
+        next_lib = todo.pop()
+        if next_lib.id in did:
+            continue
+        else:
+            import_stack.append(next_lib)
+        while import_stack:
+            if import_stack[-1].dependence:
+                if import_stack[-1].dependence in did:
+                    lib = resolve_library(import_stack[-1], [output_lib])
+
+                    output_lib.models.update(lib.models)
+                    output_lib.port_types.update(lib.port_types)
+
+                    did.append(import_stack.pop().id)
+                elif yaml_libraries[import_stack[-1].dependence] in import_stack:
+                    raise Exception("importing loop in yaml libraries")
+                else:
+                    import_stack.append(yaml_libraries[import_stack[-1].dependence])
+            else:
+                lib = resolve_library(import_stack[-1], [output_lib])
+
+                output_lib.models.update(lib.models)
+                output_lib.port_types.update(lib.port_types)
+
+                did.append(import_stack.pop().id)
+    return output_lib
+
+
+def input_database(study_path: Path, timeseries_path: Optional[Path]) -> DataBase:
+    with study_path.open() as comp:
+        return build_data_base(parse_yaml_components(comp), timeseries_path)
+
+
+def input_components(study_path: Path, model: Library) -> NetworkComponents:
+    with study_path.open() as comp:
+        return resolve_components_and_cnx(parse_yaml_components(comp), model)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--study", type=Path, help="path to the root dirertory of the study"
+    )
+    parser.add_argument(
+        "--models", nargs="+", type=Path, help="list of path to model file, *.yml"
+    )
+    parser.add_argument(
+        "--component", type=Path, help="path to the component file, *.yml"
+    )
+    parser.add_argument(
+        "--timeseries", type=Path, help="path to the timeseries dirertory"
+    )
+    parser.add_argument(
+        "--duration", type=int, help="duration of the simulation", default=1
+    )
+    parser.add_argument(
+        "--scenario", type=int, help="number of scenario of the simulation", default=1
+    )
+
+    args = parser.parse_args()
+
+    components = None
+    database = None
+
+    if args.study:
+        if args.models or args.component or args.timeseries:
+            parser.error(
+                "--study flag can't be use with --models, --component and --timeseries"
+            )
+        components_path = args.study / "input" / "components" / "components.yml"
```

Comment:
Let's move this logic (of parsing all data from one directory) to `andromede.study.parsing`